text/0374-jsii-ts-version.md

```diff
@@ -0,0 +1,504 @@
+# The `jsii` compiler (and `jsii-rosetta`) to follow TypeScript versioning
+
+* **Original Author(s)**: @RomainMuller
+* **Tracking Issue**: #374
+
+This RFC proposes to change the versioning scheme of the `jsii` compiler and
+`jsii-rosetta` to stop conforming to [semantic versioning][semver], and instead
+use the `major.minor` version of the [TypeScript] compiler it is built on.
+
+Other packages (such as `jsii-pacmak`, `@jsii/spec`, ...) are not affected by
+this proposal and will continue to conform to [semantic versioning][semver]. In
+particular, code generated by `jsii-pamcak` will continue to use the `1.x`
+release line of `@jsii/kernel`, `@jsii/runtime`, and the languages' runtime
+libraries, so as to remain completely backwards compatible.
+
+## Working Backwards
+
+### Release Notes entry
+
+```markdown
+Starting with this release, the `jsii` compiler and `jsii-rosetta` will no
```

Comment:
As a final nit, the emphasis on "semantic versioning" in this document feels a little misplaced.

Ultimately the crux of it is that there will be two different *versioning lines* or *versioning strategies* used in the jsii ecosystem:

- One for the spec and runtime, which is our own versioning line
- One for the user-facing tools that process TypeScript, which follows `tsc` versioning line

It just so happens that our controlled versioning line will follow semver and the other one will not.

But even **more importantly**, I suppose we will also guarantee eternal backwards compatibility in our own versioning line? (Which makes the discussion about semver a bit moot, effectively it becomes "we will never break backwards compatibility AND we version according to semver ⇒ our major version will always remain 1")

But the different versioning lines and our backwards compatibility promise (if any) should be more prominent than whether we force some technical standard on our version number.

Reply:
Thats a very interesting way to put it, and I like it. I was perhaps excessively focusing on the change of expectations users can have about the maintenance & compatibility of various versions of the tools (i.e: so far we claimed to do `semver`, and we may no longer do in some spaces). But that doesn't do a great job at expressing what is the new state of the world in a clear way.

I'll look into re-phrasing those bits...